### PR TITLE
docs(stack): expands design guidelines for stack layout.

### LINKED ISSUE #3588: Enhance design guidelines for our Layouts pages - Stack
Add guidelines to Stack Layouts page
Include picture example of Stack being used

### PR DESCRIPTION
**Pull request by edonehoo**
closes #3588

### CODE REVIEW

**Comment by edonehoo** [comment]
I'm currently using placeholder images for examples. Need to get real UI images (or at least mock ups) from somewhere. 

LMK if the language around allowing multiple items to fill is correct. This is what I gathered from the notes I had, but I know the original stack description only mentioned one item filling.

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md [hunk 1]
```diff
@@ -2,4 +2,31 @@
...
+
+### When to use stack vs other layouts 
+
+The stack layout does not wrap. If you need a layout that wraps, instead choose the flex, gallery, level, or split layout instead.
+
+## Variations
+
+### Multiple items that fill space 
+
+You can specifiy which stack item fill the remaining vertical space in the layout. More than one items can be set to fill, and they will share the remaining space equally.
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md:24]:
```suggestion
You can specify which stack item fill the remaining vertical space in the layout. More than one item can be set to fill and they will share the remaining space equally. 
```

**Comment by edonehoo** [comment]
@lboehling @mmenestr

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md [hunk 2]
```diff
@@ -2,4 +2,31 @@
...
+<img src="./img/stack-example.png"  alt="PatternFly website screenshot showing modal created with bullseye layout."  width="3330px"/>
+
+1. **Stack**: The stack container, which holds all stack items.
+1. **Stack item**: An individual section of content to be placed in the stack. 
+
+## Usage
+
+### When to use stack vs other layouts 
+
+The stack layout does not wrap. If you need a layout that wraps, instead choose the flex, gallery, level, or split layout instead.
```
Comment (mmenestr) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md:18]: I feel like the way this is structured doesn't align well with the header. I'd say something like "User the stack layout when you do not need your layout to wrap. If you need a layout that wraps, instead.. etc"

File: packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md [hunk 3]
```diff
@@ -2,4 +2,31 @@
...
+
+## Variations
+
+### Multiple items that fill space 
+
+You can specify which stack item fill the remaining vertical space in the layout. More than one item can be set to fill and they will share the remaining space equally.
```
Comment (mmenestr) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/layouts/stack.md:24]: Should this say which stack item fill**S**, not item fill? or am I reading it wrong

**Review by mmenestr — CHANGES_REQUESTED** [review]
left a couple comments!

**Review by mmenestr — APPROVED** [review]

**Pull request merged**